v2/tools/asoctl/internal/crd/cleaner_test.go

```diff
@@ -0,0 +1,311 @@
+	fakeApiExtClient := fake.NewSimpleClientset().ApiextensionsV1()
+	fakeClient := fake2.NewClientBuilder().WithScheme(controllers.CreateScheme()).Build()
+	cleaner := NewCleaner(fakeApiExtClient.CustomResourceDefinitions(), fakeClient, false)
+	return fakeApiExtClient, fakeClient, cleaner
+}
+
+func Test_CleanDeprecatedCRDVersions_CleansAlphaVersion_IfExists(t *testing.T) {
+	t.Parallel()
+
+	fakeApiExtClient, _, cleaner := getClientSets()
+
+	asserter := NewGomegaWithT(t)
+
+	alphaVersion := "v1alpha1api20200601"
+	betaVersion := "v1beta20200601"
+
+	definition := newCRDWithStoredVersions("v1alpha1api20200601", "v1beta20200601")
+
+	_, err := fakeApiExtClient.CustomResourceDefinitions().Create(context.TODO(), definition, metav1.CreateOptions{})
+	asserter.Expect(err).To(BeNil())
+
+	err = cleaner.Run(context.TODO())
+	asserter.Expect(err).To(BeNil())
+
+	crd, err := fakeApiExtClient.CustomResourceDefinitions().Get(context.TODO(), definition.Name, metav1.GetOptions{})
+	asserter.Expect(err).To(BeNil())
+
+	asserter.Expect(crd.Status.StoredVersions).ToNot(BeNil())
+	asserter.Expect(crd.Status.StoredVersions).ToNot(BeEquivalentTo(definition.Status.StoredVersions))
+	asserter.Expect(crd.Status.StoredVersions).ToNot(ContainElement(alphaVersion))
+	asserter.Expect(crd.Status.StoredVersions).To(ContainElement(betaVersion))
+}
+
+func Test_MigrateDeprecatedCRDResources_MigrateAlphaVersion_IfStorage(t *testing.T) {
+	t.Parallel()
+	asserter := NewGomegaWithT(t)
+
+	fakeApiExtClient, fakeClient, cleaner := getClientSets()
+
+	alphaVersion := "v1alpha1api20200601"
+	betaVersion := "v1beta20200601"
+
+	// create CRD
+	definition := newCRDWithStoredVersions(alphaVersion, betaVersion)
+	definition.Spec.Versions = []v1.CustomResourceDefinitionVersion{
+		{
+			Name:    alphaVersion,
+			Storage: true,
+		},
+		{
+			Name: betaVersion,
+		},
+	}
+
+	ns := newNamespace("test-ns")
+
+	_, err := fakeApiExtClient.CustomResourceDefinitions().Create(context.TODO(), definition, metav1.CreateOptions{})
+	asserter.Expect(err).To(BeNil())
+
+	// create Namespace
+	err = fakeClient.Create(context.TODO(), ns)
+	asserter.Expect(err).To(BeNil())
+
+	// create ResourceGroup
+	rg := newResourceGroup("test-rg", ns.Name)
+	err = fakeClient.Create(context.TODO(), rg)
+	asserter.Expect(err).To(BeNil())
+
+	err = cleaner.Run(context.TODO())
+	asserter.Expect(err).To(BeNil())
+
+	crd, err := fakeApiExtClient.CustomResourceDefinitions().Get(context.TODO(), definition.Name, metav1.GetOptions{})
+	asserter.Expect(err).To(BeNil())
+
+	var updatedRG resources.ResourceGroup
+	err = fakeClient.Get(context.TODO(), types.NamespacedName{Name: rg.Name, Namespace: rg.Namespace}, &updatedRG)
+	asserter.Expect(err).To(BeNil())
+	asserter.Expect(updatedRG.ResourceVersion).ToNot(BeEquivalentTo(rg.ResourceVersion))
+
+	asserter.Expect(crd.Status.StoredVersions).ToNot(BeNil())
+	asserter.Expect(crd.Status.StoredVersions).ToNot(BeEquivalentTo(definition.Status.StoredVersions))
```

Comment:
minor: I think this 1 assertion is technically not needed. Just by asserting that it doesn't contain alpha you sorta imply that it isn't equal to `definition.Status.StoredVersions`